_posts/2025-04-19-navigating-pagination-with-hybrid-query.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: post
+title:  "Navigating pagination in hybrid queries with the pagination_depth parameter"
+authors:
+- varunudr
+- minalsha
+- seanzheng
+- kolchfa
+date: 2025-04-16
+categories:
+  - technical-posts
+meta_keywords: hybrid queries, semantic search, lexical search, search relevance, query execution workflow, OpenSearch hybrid search, Pagination
+meta_description: Explore how pagination works in hybrid queries by using pagination depth to efficiently paginate over the result set.
+---
+
+OpenSearch 2.10 introduced hybrid queries, which have become a popular choice for improving semantic search relevance. By combining full-text lexical search with semantic search, hybrid queries deliver better results than either method alone across various applications, including e-commerce, document search, log analytics, and data exploration. Our [earlier blog post](https://opensearch.org/blog/hybrid-search/) introduced this feature and presented quality and performance results.
+
+OpenSearch continues to enhance its hybrid query capabilities with features like post-filters, aggregations, query parallelization, the `explain` parameter, and sorting. Building on this momentum, OpenSearch 2.19 introduced pagination support through the new `pagination_depth` parameter. When used with traditional `from` and `size` parameters, this enhancement helps you efficiently navigate through hybrid query result sets. For example, you can now paginate through thousands of search results one page at a time.
```

Comment:
Add a link to OpenSearch 2.19 here: https://opensearch.org/blog/explore-opensearch-2-19/